llm/server.go

```diff
@@ -21,32 +21,57 @@ import (
 	"strings"
 	"time"
 
+	"golang.org/x/sync/semaphore"
+
 	"github.com/ollama/ollama/api"
 	"github.com/ollama/ollama/format"
 	"github.com/ollama/ollama/gpu"
 )
 
-// LlamaServer is an instance of the llama.cpp server
-type LlamaServer struct {
+type LlamaServer interface {
```

Comment:
might be a silly question but why the re-introduction of an interface? for testing?

Reply:
disregard - saw the `mockLLM` further in the diff!